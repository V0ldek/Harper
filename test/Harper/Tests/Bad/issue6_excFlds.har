value List a = {
  variant Empty = { };
  
  variant NonEmpty = {
    data = {
      head :: a;
      tail :: List a;
    }
  };
};

cons x xs = val NonEmpty { head = x, tail = xs };

empty = val Empty {};

main = {
  list = val Empty { head = 5 };
  return list;
};