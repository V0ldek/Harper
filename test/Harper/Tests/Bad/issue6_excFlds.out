Error: unrecognized field identifiers during value construction. Type `List.Empty` has no fields: `head`.
During evaluation of:
 val Empty { head = 5 }

Located at line 17 column 10

Execution terminated with an error: runtime error.